### PR TITLE
vic-machine delete integration test

### LINKED ISSUE #1228: Add integration test for vic-machine delete

### PR DESCRIPTION
**Pull request by emlin**
Also support default value for "compute-resource" and "name" in vic-machine delete, to make it consistent with vic-machine create

Fixes #1228

### CODE REVIEW

File: cmd/vic-machine/delete/delete.go [hunk 1]
```diff
@@ -90,11 +90,12 @@ func (d *Uninstall) processParams() error {
 
 	if d.id != "" {
 		log.Warnf("ID of Virtual Container Host is not supported until vic-machine ls is ready. For details, please refer github issue #810")
+		return cli.NewExitError("must specify --compute-resource and --name", 1)
 	}
 
-	if (d.ComputeResourcePath == "" || d.DisplayName == "") && d.id == "" {
-		return cli.NewExitError("must specify --vch-id, or both --compute-resource and --name", 1)
-	}
+	//	if (d.ComputeResourcePath == "" || d.DisplayName == "") && d.id == "" {
```
Comment (mdubya66) [inline cmd/vic-machine/delete/delete.go]: why comment out instead of delete?

Comment (emlin) [inline cmd/vic-machine/delete/delete.go]: Good point. Comment out is for test only.

File: tests/test-cases/Group6-VIC-Machine/6-1-Help.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Test 2-4 - Verify Help
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-1-Help.md]: Test 6-1 - Verify Help

File: tests/test-cases/Group6-VIC-Machine/6-2-Default.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Test 2-4 - Verify Help
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-2-Default.md]: Test 6-2 - Verify Default

File: tests/test-cases/Group6-VIC-Machine/6-2-Default.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+=======
+
+#Purpose:
+Verify vic-machine delete default parameters of compute-resource and name
+
+#References:
+* vic-machine-linux delete --target XXX
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-2-Default.md]: The reference for this should probably still just be vic-machine-linux delete -h, as that is where the options are documented.

File: tests/test-cases/Group6-VIC-Machine/6-2-Default.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+#Environment:
+Standalone test requires nothing but vic-machine to be built
+
+#Test Steps:
+1. Issue the following command:
+* vic-machine-linux delete --target XXX
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-2-Default.md]: Triple tic this command (```)

File: tests/test-cases/Group6-VIC-Machine/6-2-Default.md (hunk 4 above)
Comment (emlin) [inline tests/test-cases/Group6-VIC-Machine/6-2-Default.md]: Sorry for forgot to update the title. But this is to test default parameters. Will update the title.

File: tests/test-cases/Group6-VIC-Machine/6-2-Default.robot [hunk 6]
```diff
@@ -0,0 +1,8 @@
+*** Settings ***
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-2-Default.robot]: Documentation  Test 6-2 - Verify Default

File: tests/test-cases/Group6-VIC-Machine/6-1-Help.robot [hunk 7]
```diff
@@ -0,0 +1,7 @@
+*** Settings ***
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-1-Help.robot]: Documentation  Test 6-1 - Verify Help

File: tests/test-cases/Group6-VIC-Machine/6-2-Default.robot [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+Resource  ../../resources/Util.robot
+
+*** Test Cases ***
+Test
+    ${ret}=  Run  bin/vic-machine-linux delete --target %{TEST_URL} --user %{TEST_USERNAME} --password=%{TEST_PASSWORD}
+    Should Contain  ${ret}  vic-machine-linux failed:  resource pool
+	Should Contain  ${ret}  /Resources/docker-appliance' not found
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-2-Default.robot]: Extra tab in here at the beginning

File: tests/test-cases/Group6-VIC-Machine/6-3-Delete.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
+Test 2-4 - Verify Help
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-3-Delete.md]: Test 6-3 - Verify Delete

File: tests/test-cases/Group6-VIC-Machine/6-3-Delete.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+=======
+
+#Purpose:
+Verify vic-machine delete can delete vch installed by vic-machine create
+
+#References:
+* vic-machine-linux delete --target XXX --user XXX --password XXX --compute-resource <resource pool used by create> --name <vch name used by create>
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-3-Delete.md]: Reference should be -help again here, as that is where the opts are documented.

File: tests/test-cases/Group6-VIC-Machine/6-3-Delete.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+#Environment:
+Standalone test requires nothing but vic-machine to be built
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-3-Delete.md]: This actually needs a vSphere instance to run

File: tests/test-cases/Group6-VIC-Machine/6-2-Default.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+=======
+
+#Purpose:
+Verify vic-machine delete default parameters of compute-resource and name
+
+#References:
+* vic-machine-linux delete --target XXX
+
+#Environment:
+Standalone test requires nothing but vic-machine to be built
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-2-Default.md]: This one need a vSphere instance to run the test

File: tests/test-cases/Group6-VIC-Machine/6-3-Delete.robot [hunk 13]
```diff
@@ -0,0 +1,42 @@
+*** Settings ***
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-3-Delete.robot]: Documentation Test 6-3 - Verify Delete

File: tests/test-cases/Group6-VIC-Machine/6-3-Delete.robot [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    ${ret}=  Run  govc vm.info -json=true ${vch-name}
+    Should Contain  ${ret}  {"VirtualMachines":null}
+
+    # Check directories is removed
+    ${ret}=  Run  govc datastore.ls VIC/${uuid}
+    Should Contain  ${ret}   was not found
+    ${ret}=  Run  govc datastore.ls ${vch-name}
+    Should Contain  ${ret}   was not found
+    ${ret}=  Run  govc datastore.ls VIC/${container-id}
+    Should Contain  ${ret}   was not found
```
Comment (mhagen-vmware) [inline tests/test-cases/Group6-VIC-Machine/6-3-Delete.robot]: We should check that the resource pool is removed as well, if we support that currently, something like this:
${ret}=  Run  govc pool.info host/*/Resources/${vch-name}
Should Contain  ${ret}  was not found

**Comment by andrewtchin** [comment]
lgtm

**Comment by mhagen-vmware** [comment]
lgtm

**Comment by mhagen-vmware** [comment]
.

**Comment by emlin** [comment]
refresh

**Comment by mdubya66** [comment]
lgtm

**Comment by emlin** [comment]
Killed build for just did branch update. Here is the passed test: https://ci.vmware.run/vmware/vic/2353

**Pull request merged**